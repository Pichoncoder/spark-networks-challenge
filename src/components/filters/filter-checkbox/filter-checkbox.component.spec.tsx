import React from 'react'
import TestRenderer, { create } from 'react-test-renderer';
import FilterCheckbox from './filter-checkbox.component';
import { shallow, mount} from "enzyme";
import FilterModule from '../../../App';

const props: any = { text: "Photo", filter: 'has_photo', handleCheckbox: jest.fn() };

describe('<FilterCheckbox /> spec', () => {
  it('Snapshot:renders the component', () => {
      const container = TestRenderer.create(<FilterCheckbox {...props}></FilterCheckbox>);
      expect(container).toMatchSnapshot();
  }); 

  it('1#Click on checkbox and state update its value', () => {
    const wrapper = shallow(<FilterCheckbox {...props} />)

    wrapper.find('.filter-element__checkbox').simulate('change', {
      target: true,
  });

    expect(wrapper).toMatchSnapshot()
  });

  it('2#Click on checkbox and state update its value to false', () => {
    const wrapper = shallow(<FilterCheckbox {...props} />)
    wrapper.find('input').simulate('change');

    const parent = mount(<FilterModule></FilterModule>);
    console.log(wrapper.state());

   // expect(wrapper.find('.filter-element__checkbox').prop('checked')).toBe(true)
    expect(props.handleCheckbox).toBeCalledWith('has_photo', true);
  });
});