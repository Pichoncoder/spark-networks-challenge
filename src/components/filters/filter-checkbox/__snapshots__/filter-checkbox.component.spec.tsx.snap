// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterCheckbox /> spec 1#Click on checkbox and state update its value 1`] = `
<li
  className="filter-element checkbox-filter has_photo"
>
  <label
    className="filter-element__text"
  >
     
    Photo
    <input
      className="filter-element__checkbox"
      id="has_photo"
      name="has_photo"
      onChange={[Function]}
      type="checkbox"
    />
  </label>
</li>
`;

exports[`<FilterCheckbox /> spec Snapshot:renders the component 1`] = `
<li
  className="filter-element checkbox-filter has_photo"
>
  <label
    className="filter-element__text"
  >
     
    Photo
    <input
      className="filter-element__checkbox"
      id="has_photo"
      name="has_photo"
      onChange={[Function]}
      type="checkbox"
    />
  </label>
</li>
`;
